Improved PHP-FPM diagnostic check config option compatibility

bx80 opened the pull request:
### Description:

The additional access rule checks for PHP-FPM added by #18398 do not respect the `enable_required_directories diagnostic` `config.ini.php` setting and are therefore causing unwanted http requests for config files when the setting is set to disabled.

This PR adjusts this diagnostic check to only attempt file access if the config setting allows it.

Fixes #18967

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

Commit:
Adjust PHP-FPM diagnostic check to respect enable_required_directories_diagnostic config setting

Commit:
Updated UI test screenshots

sgiehl commented on plugins/Diagnostics/Diagnostic/PhpInformational.php:
```diff
@@ -38,38 +39,79 @@ public function execute()
+
+        // Check for php fpm
         $isGlobalConfigIniAccessible = true; // Assume true if not installed yet
+        $disableDirectoryCheck = (Config::getInstance()->General['enable_required_directories_diagnostic'] == 0);
```
We should try to use `GeneralConfig::getConfigValue()` instead of `Config::getInstance()->General`

sgiehl reviewed (changes requested):
Actually I was wondering if we really need to make it that complex. If some private directories are accessible, I guess the other diagnostic check will print a warning nevertheless, so there isn't much value in adding another warning here.
I guess I would simply skip the directory check here, if disabled in config and print the sapi name

Commit:
Simplified warning message display logic, now using GeneralConfig::getConfigValue

bx80 commented:
I've simplified it so that the file access warning message will only be shown if they are using PHP-FPM and the `enable_required_directories_diagnostic` is not disabled and the file is accessible, for all other scenarios it will just show the sapi name. Also switched to using the `GeneralConfig` class.

justinvelluppillai commented on plugins/Diagnostics/Diagnostic/PhpInformational.php:
```diff
@@ -8,6 +8,8 @@
 namespace Piwik\Plugins\Diagnostics\Diagnostic;
 
 use Piwik\CliMulti\CliPhp;
+use Piwik\Config;
```
```suggestion
```

justinvelluppillai committed:
Remove use

Commit:
Update UI test screenshots

Commit:
Merge remote-tracking branch 'origin/m-18967-diag-fpm-respect-config' into m-18967-diag-fpm-respect-config
